File: 01/gemini-3.0-pro/go/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	file, err := os.Open("input.txt")
	if err != nil {
		panic(err)
	}
	defer file.Close()

	pos1 := 50
	count1 := 0

	// Read lines into a slice first to iterate twice
	var lines []string
	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		lines = append(lines, strings.TrimSpace(scanner.Text()))
	}

	// Part 1
	for _, line := range lines {
		if line == "" {
			continue
		}

		dir := line[0]
		amount, _ := strconv.Atoi(line[1:])

		if dir == 'R' {
			pos1 = (pos1 + amount) % 100
		} else if dir == 'L' {
			pos1 = (pos1 - amount) % 100
			if pos1 < 0 {
				pos1 += 100
			}
		}

		if pos1 == 0 {
			count1++
		}
	}

	fmt.Printf("Day 01 Part 1: %d\n", count1)

	// Part 2
	pos2 := 50
	count2 := 0

	for _, line := range lines {
		if line == "" {
			continue
		}

		dir := line[0]
		amount, _ := strconv.Atoi(line[1:])

		for i := 0; i < amount; i++ {
			if dir == 'R' {
				pos2 = (pos2 + 1) % 100
			} else if dir == 'L' {
				pos2--
				if pos2 < 0 {
					pos2 = 99
				}
			}

			if pos2 == 0 {
				count2++
			}
		}
	}

	fmt.Printf("Day 01 Part 2: %d\n", count2)
}

File: 01/gemini-3.0-pro/rust/src/main.rs
use std::fs;

fn main() {
    let input = fs::read_to_string("input.txt").expect("Unable to read file");

    // Part 1
    let mut pos1 = 50;
    let mut count1 = 0;

    for line in input.lines() {
        let line = line.trim();
        if line.is_empty() {
            continue;
        }

        let dir = line.chars().next().unwrap();
        let amount: i32 = line[1..].parse().unwrap();

        if dir == 'R' {
            pos1 = (pos1 + amount) % 100;
        } else if dir == 'L' {
            pos1 = (pos1 - amount) % 100;
            if pos1 < 0 {
                pos1 += 100;
            }
        }

        if pos1 == 0 {
            count1 += 1;
        }
    }

    println!("Day 01 Part 1: {}", count1);

    // Part 2
    let mut pos2 = 50;
    let mut count2 = 0;

    for line in input.lines() {
        let line = line.trim();
        if line.is_empty() {
            continue;
        }

        let dir = line.chars().next().unwrap();
        let amount: i32 = line[1..].parse().unwrap();

        for _ in 0..amount {
            if dir == 'R' {
                pos2 = (pos2 + 1) % 100;
            } else if dir == 'L' {
                pos2 -= 1;
                if pos2 < 0 {
                    pos2 = 99;
                }
            }

            if pos2 == 0 {
                count2 += 1;
            }
        }
    }

    println!("Day 01 Part 2: {}", count2);
}
